internal/access/signup.go

```diff
@@ -11,50 +11,29 @@ import (
 	"github.com/infrahq/infra/uid"
 )
 
-// SignupEnabled queries the current state of the service and returns whether
-// or not to allow unauthenticated signup. Signup is enabled if and only if
-// the configuration flag 'enableSignup' is set and no identities, providers,
-// or grants have been configured, both currently and previously.
-func SignupEnabled(c *gin.Context) (bool, error) {
+// Signup creates a user identity using the supplied name and password and
+// grants the identity "admin" access to Infra.
+func Signup(c *gin.Context, name, password, orgName string) (*models.Identity, error) {
 	// no authorization is setup yet
 	db := getDB(c)
 
-	// use Unscoped because deleting identities, providers or grants should not re-enable signup
-	identities, err := data.Count[models.Identity](db.Unscoped(), data.NotName(models.InternalInfraConnectorIdentityName))
-	if err != nil {
-		return false, err
-	}
-
-	providers, err := data.Count[models.Provider](db.Unscoped(), data.NotProviderKind(models.ProviderKindInfra))
-	if err != nil {
-		return false, err
-	}
-
-	grants, err := data.Count[models.Grant](db.Unscoped(), data.NotPrivilege(models.InfraConnectorRole))
-	if err != nil {
-		return false, err
-	}
-
-	accessKeys, err := data.Count[models.AccessKey](db.Unscoped())
+	org := &models.Organization{Name: orgName}
+	err := data.CreateOrganization(db, org)
 	if err != nil {
-		return false, err
+		return nil, err
 	}
 
-	return identities+providers+grants+accessKeys == 0, nil
-}
-
-// Signup creates a user identity using the supplied name and password and
-// grants the identity "admin" access to Infra.
-func Signup(c *gin.Context, name, password string) (*models.Identity, error) {
-	// no authorization is setup yet
-	db := getDB(c)
-
-	err := checkPasswordRequirements(db, password)
+	err = checkPasswordRequirements(db, password)
 	if err != nil {
 		return nil, err
 	}
 
-	identity := &models.Identity{Name: name}
+	identity := &models.Identity{
+		Model: models.Model{
```

Comment:
Is there a simpler way to set the org ID in the current implementation than this?